pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,98 @@
+package pullsecret
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/machine/libmachine/drivers"
+	"github.com/pborman/uuid"
+)
+
+var (
+	pullSecret = `apiVersion: v1
+data:
+  .dockerconfigjson: %s
+kind: Secret
+metadata:
+  name: pull-secret
+  namespace: openshift-config
+type: kubernetes.io/dockerconfigjson`
+
+	// This command timeout in 80 secs if not able to replace the pull
+	// secret. We first start the kubelet and wait till the api server is
+	// up and running by hitting it with replace command.
+	replacePullSecretCmd = `timeout 80 bash -c 'sudo systemctl start kubelet && \
```

Comment:
Hiding `systemctl start kubelet` in `replacePullSecretCmd` is a bit opaque, as this means we can't move `updateClusterIdCmd` first, and then call `replacePullSecretCmd`. I'd start the kubelet explicitly using `systemd.NewInstanceSystemdCommander`

Reply:
@cfergeau why we want move `updateClusterIdCmd` first, Yes we can have this command separate this command.

Reply:
No need to move it, this was just an example to explain why I think it's better to separate kubelet startup from pull secret update.

Reply:
@cfergeau got you, now it is separated.